apps/framework-cli/src/framework/transform.ts

```diff
@@ -102,20 +107,27 @@ const handleMessage = async (
           topic: `${destination}_${version}`,
           messages: [{ value: JSON.stringify(transformedData) }],
         });
-        await transaction.commit();
+        didTransform = true;
         console.log(`Sent transformed data to ${destination}`);
-      } else {
-        transaction.abort();
       }
+    }
 
-      resolveOffset(message.offset);
-    } catch (error) {
+    if (didTransform) {
+      await transaction.sendOffsets({
```

Comment:
looks like now we have exactly-once processing? This is too great to not be in the title.

Reply:
Haha, I appreciate you adding the TODO there & providing guidance.